libraries/AP_ADSB/AP_ADSB.cpp

```diff
@@ -639,7 +639,7 @@ uint8_t AP_ADSB::get_encoded_callsign_null_char()
 
     // using the above logic, we must always assign the squawk. once we get configured
     // externally then get_encoded_callsign_null_char() stops getting called
-    snprintf(out_state.cfg.callsign, 5, "%04d", unsigned(out_state.cfg.squawk_octal));
+    snprintf(out_state.cfg.callsign, 5, "%04d", unsigned(out_state.cfg.squawk_octal) & 0x001F);
```

Comment:
The approach outlined works, but as @amilcarlucas points out the mask selected here is definitely wrong.